lib/modules/manager/terragrunt/artifacts.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { join } from 'upath';
+import { mocked } from '../../../../test/util';
+import { GlobalConfig } from '../../../config/global';
+import type { UpdateType } from '../../../config/types';
+import * as TerraformLockfile from '../terraform/lockfile';
+import type { UpdateArtifactsConfig } from '../types';
+import { updateArtifacts } from './artifacts';
+
+jest.mock('../terraform/lockfile');
+const mockTFUpdateArtifacts = mocked(TerraformLockfile).updateArtifacts;
+
+const config = {
+  constraints: {},
+};
+
+const adminConfig = {
+  // `join` fixes Windows CI
+  localDir: join('/tmp/github/some/repo'),
+  cacheDir: join('/tmp/renovate/cache'),
+  containerbaseDir: join('/tmp/renovate/cache/containerbase'),
+};
+
+describe('modules/manager/terragrunt/artifacts', () => {
+  const updateTypes: UpdateType[] = [
+    'digest',
+    'pin',
+    'rollback',
+    'patch',
+    'minor',
+    'major',
+    'replacement',
+    'pinDigest',
+    'lockfileUpdate',
+    'bump',
+  ];
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.resetModules();
```

Comment:
```suggestion
```

the first does nothing without dynamic imports
the second does nothing, because you don't set any explicit mocks.

The mocks are cleared globally here:
https://github.com/renovatebot/renovate/blob/85fbad1029d4a10ff07b93152324b68b8a9fd990/jest.config.ts#L42